Doc/c-api/unicode.rst

```diff
@@ -1061,6 +1061,17 @@ These are the UTF-8 codec APIs:
    raised by the codec.
 
 
+.. c:function: int PyUnicode_GetUTF8Buffer(PyObject *unicode, const char errors, Py_buffer *view)
+
+   Get a read-only buffer of the UTF-8 encoding of the Unicode object.
+   Return ``-1`` if an error happened.
+
+   Successful calls to :c:func:`PyUnicode_GetUTF8Buffer` must be paired with
+   calls to :c:func:`PyBuffer_Release`, similar to :c:func:`PyObject_GetBuffer`.
+
```

Comment:
Since the motivation to add this new function was performance, would you mind to elaborate a little bit on that? Explain if a memory copy is needed. Explain if the function fills the internal UTF-8 cache if it's not already computed?